src/components/store/AnalyzeStore.ts

```diff
@@ -16,44 +17,75 @@ interface AnalyzeActions {
 	}
 }
 
-// 카테고리별 뭉치고 가격 합산
-const combiningData = (
-	consumptionData: analyzeDataType[],
-): analyzeCombinedDataType[] => {
-	let id = 0
-	const consumptionCombinedData = Object.values(
-		consumptionData.reduce<Record<string, analyzeCombinedDataType>>(
-			(acc, item) => {
-				if (!acc[item.category]) {
-					acc[item.category] = {
-						id: id++,
-						category: item.category,
-						price: item.price,
-					}
-				} else {
-					acc[item.category].price += item.price
-				}
-				return acc
+function createDataItems(data: {
+	[key: string]: { value: number; firstId: number }
+}): spendingDataProps[] {
+	return Object.entries(data)
+		.sort(([, a], [, b]) => b.value - a.value)
+		.map(([key, { value, firstId }], index) => ({
+			id: firstId,
+			data: {
+				id: key,
+				label: key,
+				value,
+				color: chartColors[index % chartColors.length],
 			},
-			{},
-		),
-	)
-	return consumptionCombinedData.sort((a, b) => b.price - a.price)
+		}))
 }
 
+function processConsumptionData(
+	data: any[],
+	startDate: Date,
+	endDate: Date,
+): spendingDataProps[] {
+	const filteredData = data.filter((item) => {
+		const itemDate = new Date(item.date)
+		return (
+			itemDate >= startDate &&
+			itemDate <= endDate &&
+			item.input_type === 'spending'
+		)
+	})
+
+	const spendingData: { [key: string]: { value: number; firstId: number } } = {}
```

Comment:
현재 이 친구가 위에서도 이곳에서도 사용되고 있습니다. 인라인에 정의하는 것보다 `interface`로 정의해서 사용해도 괜찮을 것 같습니다.

```suggestion
interface CategorySpending {
  value: number
  firstId: number
}
...
const spendingData: { [key: string]: CategorySpending } = {}
```
또는 이렇게도 사용할 수 있습니다.
```ts
const spendingData: Record<string, CategorySpending> = {}
```
Record의 첫번째는 각 키 타입이고, 두번째는 각 키에 매핑될 값의 타입입니다.

Reply:
고수시네요, 보고 배웁니다.